app/controllers/api/v1/api_controller.rb

```diff
@@ -0,0 +1,23 @@
+class Api::V1::ApiController < ActionController::API
+  before_action :set_default_format
+  before_action :authenticate_token!
+
+   private
+
+     def set_default_format
+      request.format = :json
+    end
+
+    def authenticate_token!
+      payload = JsonWebToken.decode(auth_token)
+      @current_user = User.find(payload["sub"])
+      rescue JWT::ExpiredSignature
+        render json: {errors: ["Auth token has expired"]}, status: :unauthorized
+      rescue JWT::DecodeError
+        render json: {errors: ["Invalid auth token"]}, status: :unauthorized
+    end
```

Comment:
again spacing seems to be out of synch.

Reply:
Sir it not out of sync. On rescue it catches exception, then executes next line.